sympy/printing/codeprinter.py

```diff
@@ -134,8 +134,19 @@ def _handle_assign_to(expr, assign_to):
                         type(self).__name__, type(assign_to)))
             return Assignment(assign_to, expr)
 
+        #for codeprinting, all lists should be converted to abstract_nodes.Lists.
+        from sympy.codegen.abstract_nodes import List
+        store_list_converter=converter.get(list, None)
+        converter[list]=lambda x: List(*x)
```

Comment:
I'm adding `list->List` conversion to `sympify` for the purposes of `CodePrinting`, this will allow current tuple/list nesting without creating `list`s in `Tuple`s but not change anything for the user. Is there a better way to do this?

Reply:
This kind of coding is fragile in my humble opinion: importing a mutable object, modifying it in-place, calling a function using it, and then reverting the change. If the `converter` in `.sympify` is intended to be modifiable, then I'd rather see a context manager class in `.sympify` allowing to override the default converters for a region of code, would you agree?

Reply:
This is also not thread-safe. The converters are only really supposed to be registered during start up.

Probably somewhere there should explicit code that checks for `list` and converts to `List` rather than depending on `sympify` to do this implicitly.

Reply:
Point taken, I'll try to see of sympy.strategies can be of any help.

Reply:
I think this has been solved.